Lib/test/libregrtest/main.py

```diff
@@ -433,26 +433,23 @@ def display_header(self):
         print("== encodings: locale=%s, FS=%s"
               % (locale.getpreferredencoding(False),
                  sys.getfilesystemencoding()))
-        rl_version_cmd = """if 1:
-        try:
-            import readline
-        except ImportError:
-            readline = None
-        if readline is not None:
-            print(f"{readline._READLINE_VERSION:#06x}")
-            is_editline = readline.__doc__ and "libedit" in readline.__doc__
-            print("editline" if is_editline else "GNU readline")
-        """
-        rl_output = subprocess.run([sys.executable, "-c", textwrap.dedent(rl_version_cmd)],
-                                   stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        if rl_output.returncode == 0:
+        rl_version_cmd = textwrap.dedent("""
+            import sys
             try:
-                rl_version, rl_implementation = rl_output.stdout.decode().splitlines()
-            except ValueError:
-                pass
+                import readline
+            except ImportError:
+                sys.exit(1)
             else:
```

Comment:
We can remove the try/else/exit, just import readline:

    import readline

If import fails, we exit with exit code 1 anyway, no special code needed.